pkg/servicelb/controller.go

```diff
@@ -218,7 +220,9 @@ func (h *handler) podIPs(pods []*core.Pod) ([]string, error) {
 }
 
 func (h *handler) deployPod(svc *core.Service) error {
-
+	if err := h.deleteOldDeployments(svc); err != nil {
```

Comment:
Can you use the cache here?  Check if you can find the old deployment in the cache and then delete it?  I'd like to avoid an necessary REST call here.

Reply:
sure will fix

Reply:
fixed .. https://github.com/rancher/k3s/pull/435/files#diff-e679d7067357879a81950954e026625fR375